java/src/main/java/eu/solven/cleanthat/engine/java/refactorer/mutators/UnnecessaryModifier.java

```diff
@@ -141,6 +141,10 @@ protected boolean processNotRecursively(NodeAndSymbolSolver<?> nodeAndSymbolSolv
 				if (isStatic) {
 					return false;
 				}
+				if (isFinal && ((MethodDeclaration) parentNode).isPrivate()
```

Comment:
Given previous comment, I feel such an additional case is better tested with a `@CompareInnerClasses` test-case (if possible of course, which I feel is the case here).